packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -2212,8 +2212,16 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
       _lastFormattedValue = value;
     }
 
-    // Setting _value here ensures the selection and composing region info is passed.
-    _value = value;
+    if (value == _value) {
+      // Please note that the setter will not notify subscribers if set duplicate values.
+      // If the value was modified by the formatter, the remote should be notified to keep in sync,
+      // if not modified, it will short-circuit.
+      _updateRemoteEditingValueIfNeeded();
```

Comment:
This function is manually called only when the value has not changed. In other cases, the controller's notifier will call it.